airflow/models/dag.py

```diff
@@ -1782,6 +1787,9 @@ def create_dagrun(
         :meta private:
         """
         logical_date = timezone.coerce_datetime(logical_date)
+        # For manual runs where logical_date is None, ensure no data_interval is set.
+        if logical_date is None:
+            data_interval = None
```

Comment:
This should be ensured at the _call site_. This function can raise an error if data interval is provided when logical date is None, although I think even that’s not necessary. It does not make sense, but we can allow it.

Reply:
Changed it. I am raising error now.